### PR TITLE
[#3652] attempt to fix NPE

### PR DESCRIPTION
**Pull request by valllllll2000**
### Before the PR (what is the issue or what needed to be done)
NA
#### The solution
Fixed NPE when missing registration form submission for a given datapoint
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

File: GAE/src/org/akvo/flow/api/app/DataPointUtil.java [hunk 1]
```diff
@@ -35,10 +33,18 @@
 import org.waterforpeople.mapping.domain.SurveyInstance;
 import org.waterforpeople.mapping.serialization.response.MediaResponse;
 
-import com.gallatinsystems.survey.dao.QuestionDao;
-import com.gallatinsystems.survey.domain.Question;
-import com.gallatinsystems.surveyal.domain.SurveyedLocale;
 import javax.annotation.Nullable;
+import java.util.ArrayList;
```
Comment (iperdomo) [inline GAE/src/org/akvo/flow/api/app/DataPointUtil.java:37]: There are lots of changes to import *style* and *order* that are not directly to the change business logic change we want.

File: GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java [hunk 2]
```diff
@@ -67,6 +44,26 @@
 import com.google.appengine.api.taskqueue.Queue;
 import com.google.appengine.api.taskqueue.QueueFactory;
 import com.google.appengine.api.taskqueue.TaskOptions;
+import org.akvo.flow.domain.DataUtils;
```
Comment (iperdomo) [inline GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java:47]: IDEM. Import re-ordering, not directly related to business logic changes.

File: GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java [hunk 3]
```diff
@@ -774,13 +771,14 @@ public SurveyInstance getRegistrationSurveyInstance(SurveyedLocale locale,
     public List<SurveyInstance> getMonitoringData(@Nonnull List<SurveyedLocale> surveyedLocales, int numberOfInstances) {
         List<SurveyInstance> result = new ArrayList<>();
 
-        surveyedLocales.forEach(surveyedLocale ->
-                result.add(getRegistrationSurveyInstance(surveyedLocale, surveyedLocale.getCreationSurveyId())));
-
-        surveyedLocales.forEach(surveyedLocale ->
+        for (SurveyedLocale s: surveyedLocales) {
```
Comment (iperdomo) [inline GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java:774]: We're changing the logic and introduce a *corner case*. We should add a test in [SurveyInstanceDAOTest](https://github.com/akvo/akvo-flow/blob/757155a918a44db85ec5f64a844cc6527decfec5/GAE/test/com/gallatinsystems/surveyal/dao/SurveyInstanceDAOTest.java)

**Review by iperdomo — CHANGES_REQUESTED** [review]
Let's add a test for this case. Where we have *data points* for a given *survey* without *Registration Form Instance*

File: GAE/src/org/akvo/flow/api/app/DataPointUtil.java (hunk 1 above)
Comment (valllllll2000) [reply GAE/src/org/akvo/flow/api/app/DataPointUtil.java:37]: the IDE does this, we need a common code style for all IDEs so this does not happen

Comment (iperdomo) [reply GAE/src/org/akvo/flow/api/app/DataPointUtil.java:37]: it does it with you reformat the whole file. We *had* a common style defined for Eclipse but not for IntelliJ IDEA
